include/sys/thread.h

```diff
@@ -104,8 +108,9 @@ typedef struct thread {
   char *td_name;   /*!< (@) name of thread */
   tid_t td_tid;    /*!< (@) thread identifier */
   /* thread state */
-  thread_state_t td_state;    /*!< (!) thread state */
-  volatile uint32_t td_flags; /*!< (!) TDF_* flags */
+  thread_state_t td_state;        /*!< (!) thread state */
+  volatile uint32_t td_flags;     /*!< (!) TDF_* flags */
+  volatile tdp_flags_t td_pflags; /*!< (p) TDP_* flags */
```

Comment:
`td_pflags` counterpart in _FreeBSD_ is private (that's what `p` in `pflags` stands for) to thread.
It looks somewhat suspicious that you need to acquire process' lock to use this field.
Can you explain?

Reply:
NetBSD's implementation uses a `l_sigrestore` field in `struct lwp`, which is protected by the thread's process lock: http://bxr.su/NetBSD/sys/kern/sys_sig.c#619
I want to set `TDP_OLDMASK` and do the mask manipulation atomically, so protecting it with `p_lock` seems reasonable, since `td_sigmask` is protected by it as well.

Reply:
Also the `p` in `pflags` here stands for "process", I wasn't aware that it stood for "private" in FreeBSD.

Reply:
IMHO it feels much more natural for `TDP_OLDMASK` bit to be a part of thread's private state. That's something that I initally expected and got curious why other choice was made.

Reply:
On second thought, `td_oldsigmask` and `TDP_OLDSIGMASK` are only accessed by the owning thread, so making them thread-private is a good idea. I have made the corresponding changes.